test/e2e/page-objects/pages/test-dapp.ts

```diff
@@ -315,28 +316,30 @@ class TestDapp {
     await this.driver.clickElement(this.revokePermissionButton);
     await this.driver.refresh();
     await this.check_pageIsLoaded();
-    await this.driver.assertElementNotPresent({
-      css: this.connectedAccount,
-      text: publicAddress.toLowerCase(),
-    });
+    await this.check_connectedAccounts(publicAddress, false);
   }
 
   /**
-   * Verifies the accounts connected to the test dapp.
+   * Check if the accounts connected to the test dapp.
    *
-   * @param connectedAccounts - The expected connected accounts separated by a comma. If no accounts are connected we can omit the param.
+   * @param connectedAccounts - Account addresses to check if connected to test dapp, separated by a comma.
+   * @param shouldBeConnected - Whether the accounts should be connected to test dapp. Defaults to true.
    */
-  async check_connectedAccounts(connectedAccounts: string = '') {
-    console.log('Verify connected accounts');
-    if (connectedAccounts) {
+  async check_connectedAccounts(
+    connectedAccounts: string,
+    shouldBeConnected: boolean = true,
+  ) {
+    if (shouldBeConnected) {
+      console.log('Verify connected accounts:', connectedAccounts);
       await this.driver.waitForSelector({
         css: this.connectedAccount,
-        text: connectedAccounts,
+        text: connectedAccounts.toLowerCase(),
       });
     } else {
-      await this.driver.waitForSelector({
+      console.log('Verify accounts not connected:', connectedAccounts);
+      await this.driver.assertElementNotPresent({
```

Comment:
i think it's safer to use `assertElementNotPresent` rather than using `waitForSelector`, because with origin implementation, it will be true whenever there's contains a space in the connected accounts address string.

Reply:
make sense, good improvement